sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -111,19 +114,29 @@ class ResolveHiveSerdeTable(session: SparkSession) extends Rule[LogicalPlan] {
   }
 }
 
-class DetermineTableStats(session: SparkSession) extends Rule[LogicalPlan] {
-  private def hiveTableWithStats(relation: HiveTableRelation): HiveTableRelation = {
-    val table = relation.tableMeta
+class DetermineTableStats(val session: SparkSession) extends Rule[LogicalPlan] {
+  private val relationToSizeMap: mutable.Map[TableIdentifier, Long] = mutable.Map.empty
+
+  private[hive] def hiveTableWithStats(relation: HiveTableRelation): HiveTableRelation = {
     val partitionCols = relation.partitionCols
     val conf = session.sessionState.conf
     // For partitioned tables, the partition directory may be outside of the table directory.
     // Which is expensive to get table size. Please see how we implemented it in the AnalyzeTable.
     val sizeInBytes = if (conf.fallBackToHdfsForStatsEnabled && partitionCols.isEmpty) {
       try {
-        val hadoopConf = session.sessionState.newHadoopConf()
-        val tablePath = new Path(table.location)
-        val fs: FileSystem = tablePath.getFileSystem(hadoopConf)
-        fs.getContentSummary(tablePath).getLength
+        val table = relation.tableMeta
+        val relationSizeMap = getRelationToSizeMap
+        if (relationSizeMap.contains(table.identifier)) {
```

Comment:
How does this rule catch up the stats updates of base relations in this approach?

Reply:
I have now handled this in a more generic way.
The way to referesh table in a Spark session is using the `REFRESH TABLE <tblname>` command or the corresponding DF apis.
With this change, the cache invalidatioon happens whenever a refresh table is invoked